File: api/src/main/java/com/meli/challenge/domain/model/Product.java
package com.meli.challenge.domain.model;

import java.math.BigDecimal;
import java.util.List;

/**
 * Immutable domain model representing a product with all the information
 * required to render a MercadoLibre style detail page.
 */
public record Product(
        String id,
        String title,
        String description,
        BigDecimal price,
        String currency,
        int stock,
        List<String> images,
        List<Variant> variants,
        PaymentOptions paymentOptions,
        Delivery delivery,
        Seller seller,
        Ratings ratings,
        List<Feature> features,
        List<Review> reviews,
        Metadata metadata) {

    public record Variant(String label, List<String> options) { }

    public record PaymentOptions(List<String> methods, String installments) { }

    public record Delivery(String type, String estimatedDate, BigDecimal cost) { }

    public record Seller(String name, double rating, int reviews, String reputation) { }

    public record Ratings(double average, int count) { }

    public record Feature(String name, String value) { }

    public record Review(String user, String comment, int rating) { }

    public record Metadata(String seoTitle, String seoDescription, String category,
                           String brand, String availability) { }
}

File: web/types/product.ts
// Types matching the Spring Boot API ProductResponseDto
export interface Product {
  id: string;
  title: string;
  description: string;
  price: number;
  currency: string;
  stock: number;
  images: string[];
  variants: Variant[];
  paymentOptions: PaymentOptions;
  delivery: Delivery;
  seller: Seller;
  ratings: Ratings;
  features: Feature[];
  reviews?: Review[]; // Optional field - backend doesn't always include it
  metadata: Metadata;
}

export interface Variant {
  label: string;
  options: string[];
}

export interface PaymentOptions {
  methods: string[];
  installments: string;
}

export interface Delivery {
  type: string;
  estimatedDate: string;
  cost: number;
}

export interface Seller {
  name: string;
  rating: number;
  reviews: number;
  reputation: string;
}

export interface Ratings {
  average: number;
  count: number;
}

export interface Feature {
  name: string;
  value: string;
}

export interface Review {
  user: string;
  rating: number;
  comment: string;
}

export interface Metadata {
  seoTitle: string;
  seoDescription: string;
  category: string;
  brand: string;
  availability: string;
}
